### PR TITLE
Updating server is now handled from the react component

### PR DESCRIPTION
**Pull request by sjoedwards**
Now emit messages to the server within the react component as opposed to the external embedded script.

To make work: 
Same steps as in readme, (don't forget to npm install!)

### CODE REVIEW

**Review by chrisgrounds — DISMISSED** [review]
Nice one!

File: src/App.js [hunk 1]
```diff
@@ -5,13 +5,35 @@ import Messages from './Components/Messages';
...
+    handleSubmit(event) {
+        event.preventDefault()
+        socket.emit('chat message', this.state.message)
+            this.setState({ message: '' })
+        }
+
+    handleMessageUpdate() {
+        socket.on('chat message', (msg) => {
+        var currentMessages = this.state.messages
+        currentMessages.push(msg);
```
Comment (chrisgrounds) [inline src/App.js]: Yeah I agree with your Slack comment about this. We should think of a better way of doing it. But for now this works.

Comment (chrisgrounds) [reply src/App.js]: Does the function body need to be indented here?

File: src/Components/Messages.jsx [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
-const Messages = (props) => <div><ul></ul></div>
+const Messages = ({messages}) => {
 
-export default Messages;
\ No newline at end of file
+    const messageLog = messages.map ( (message) => {
+        return ( <p>{message}</p>)
+    });
+
+    return (
```
Comment (chrisgrounds) [inline src/Components/Messages.jsx]: Might be a bit nicer to merge the messageLog function into the JSX, maybe? 

```JavaScript
const Messages = ({messages}) => {
    return (
        <div>
            {messages.map((message) => <p>{message}</p>)}
        </div>
    );
}
```

File: src/App.js [hunk 3]
```diff
@@ -21,8 +43,14 @@ class App extends Component {
...
                     <WelcomeMessage username={this.state.username} />
-                    <Messages />
-                    <MessageForm />
+                    <Messages
+                        messages={this.state.messages}
+                    />
+                    <MessageForm
+                        message={this.state.message}
+                        handleMessageChange={(event) => this.setState({message:event.target.value})}
+                        handleSubmit={(event) => this.handleSubmit(event)}
```
Comment (chrisgrounds) [inline src/App.js:53]: :+1: Lambda/anonymous functions FTW!

**Review by chrisgrounds — CHANGES_REQUESTED** [review]

File: src/Components/Messages.jsx (hunk 2 above)
Comment (chrisgrounds) [reply src/Components/Messages.jsx]: Up to you, I usually favour fewer lines of code, but if you think having messageLog seperate is more readable feel free to keep that.

**Review by chrisgrounds — DISMISSED** [review]

File: src/App.js (hunk 1 above)
Comment (sjoedwards) [reply src/App.js]: Yes, good spot!

File: src/Components/Messages.jsx (hunk 2 above)
Comment (sjoedwards) [reply src/Components/Messages.jsx]: This looks tidier and more logical, pushed change

**Review by chrisgrounds — APPROVED** [review]

**Pull request merged**